stake-pool/cli/src/main.rs

```diff
@@ -562,6 +562,104 @@ fn command_withdraw(
     Ok(None)
 }
 
+fn command_set_staking_auth(
+    config: &Config,
+    pool: &Pubkey,
+    stake_account: &Pubkey,
+    new_staker: &Pubkey,
+) -> CommandResult {
+    let pool_data = config.rpc_client.get_account_data(&pool)?;
+    let pool_data: StakePool = PoolState::deserialize(pool_data.as_slice())
+        .unwrap()
+        .stake_pool()
+        .unwrap();
+
+    let pool_withdraw_authority: Pubkey = PoolProcessor::authority_id(
+        &spl_stake_pool::id(),
+        pool,
+        PoolProcessor::AUTHORITY_WITHDRAW,
+        pool_data.withdraw_bump_seed,
+    )
+    .unwrap();
+
+    let mut transaction = Transaction::new_with_payer(
+        &[set_staking_authority(
+            &spl_stake_pool::id(),
+            &pool,
+            &config.owner.pubkey(),
+            &pool_withdraw_authority,
+            &stake_account,
+            &new_staker,
+            &stake_program_id(),
+        )?],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let (recent_blockhash, fee_calculator) = config.rpc_client.get_recent_blockhash()?;
+    check_fee_payer_balance(config, fee_calculator.calculate_fee(&transaction.message()))?;
+    let mut signers = vec![config.fee_payer.as_ref(), config.owner.as_ref()];
+    unique_signers!(signers);
+    transaction.sign(&signers, recent_blockhash);
+    Ok(Some(transaction))
```

Comment:
nit: this predates this PR, but a lot of this logic can probably be factored out since it's done on every command. You can see a slick of way of refactoring it in the token cli at https://github.com/solana-labs/solana-program-library/blob/master/token/cli/src/main.rs#L1926

You can save it for a separate PR though

Reply:
Agreed and yes, I'll leave it for one of the next PRs, it is better to refactor this in all CLIs at once.